samples/nrf9160/modem_shell/src/link/link.c

```diff
@@ -82,24 +82,30 @@ static int32_t modem_rsrp = LINK_RSRP_VALUE_NOT_KNOWN;
 struct ncellmeas_data {
 	struct k_work_delayable work;
 
+	struct lte_lc_ncellmeas_params params;
+
 	enum link_ncellmeas_modes mode;
-	enum lte_lc_neighbor_search_type search_type;
 	int periodic_interval;
 };
 static struct ncellmeas_data ncellmeas_work_data;
-
-static enum link_ncellmeas_modes ncellmeas_mode = LINK_NCELLMEAS_MODE_NONE;
-static enum lte_lc_neighbor_search_type ncellmeas_search_type = LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT;
-static int ncellmeas_periodic_interval;
+static struct ncellmeas_data ncellmeas_param_data = {
+	.mode = LINK_NCELLMEAS_MODE_NONE,
+	.params.search_type = LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT,
+	.periodic_interval = 0,
+};
 
 static void link_ncellmeas_worker(struct k_work *work_item)
 {
 	struct ncellmeas_data *data = CONTAINER_OF(work_item, struct ncellmeas_data, work);
+	struct lte_lc_ncellmeas_params ncellmeas_params = {
+		.gci_count = data->params.gci_count,
+		.search_type = data->params.search_type
+	};
 
 	if (data->mode == LINK_NCELLMEAS_MODE_CONTINUOUS) {
 		link_ncellmeas_start(true,
 				     data->mode,
-				     data->search_type,
+				     ncellmeas_params,
```

Comment:
Could we pass here just `data->params` and remove `ncellmeas_params`?
Or has `ncellmeas_params` been made for clarity?

Reply:
It is a container for all related parameters, mosh + lte_lc. I think that it is needed, also for clarity.

Reply:
Yes, we could pass data->params. I will change this